actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import json
+import logging
+from pathlib import Path
+from typing import List, NamedTuple, Optional, Union
+
+import aicsimageio
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+import matplotlib.image as mpimg
+import matplotlib.pyplot as plt
+import matplotlib
+
+from aics_dask_utils import DistributedHandler
+from datastep import Step, log_run_params
+
+from ...constants import DatasetFields
+from ...utils import dataset_utils
+from ..single_cell_images import SingleCellImages
+
+plt.style.use("dark_background")
+
+###############################################################################
+
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+REQUIRED_DATASET_FIELDS = [DatasetFields.CellId]
+
+
+class DiagnosticSheetResult(NamedTuple):
+    cell_id: Union[int, str]
+    diagnostic_sheet_save_path: Optional[Path] = None
+
+
+class DiagnosticSheetError(NamedTuple):
+    cell_id: Union[int, str]
+    error: str
+
+
+###############################################################################
+
+
+class MakeDiagnosticSheet(Step):
+    def __init__(self, direct_upstream_tasks: List["Step"] = [SingleCellImages]):
+        super().__init__(direct_upstream_tasks=direct_upstream_tasks)
+
+    @staticmethod
+    def _make_group_plot(
+        dataset: pd.DataFrame, metadata: str, feature: Optional[str] = None,
+    ):
+
+        # Get figure numbers, subplot numbers, and figure paths
+        figure_numbers = dataset[metadata].value_counts().index.to_numpy()
+        subplot_numbers = dataset[metadata].value_counts().values
+        figure_paths = dataset["DiagnosticSheetPath"].value_counts().index
+
+        # Index to help loop through figure numbers
+        index = 0
+
+        # Set font size
+        font = {"weight": "bold", "size": 11}
+        matplotlib.rc("font", **font)
+
+        # Loop through figure numbers
+        for i in range(len(figure_numbers)):
+
+            # Get rows and columns of figure
+            columns = int(np.sqrt(subplot_numbers[i]) + 0.5)
+            if np.sqrt(subplot_numbers[i]) != int(np.sqrt(subplot_numbers[i])):
+                rows = columns + 1
+            else:
+                rows = columns
+
+            # Set figure size
+            fig_size_width = columns * 7
+            fig_size_height = rows * 5
```

Comment:
It may be nice to have these be auto like you are doing but it may also be nice to have these be optional parameters?